tests/doc_test_framework.go

```diff
@@ -0,0 +1,255 @@
+				return walkError
+			}
+			if testCase := checkFile(path); testCase.valid {
+				testCases = append(testCases, *testCase)
+			} else if testCase.err != nil {
+				log.Printf("Error occurred while processing %v: %v", testCase.path, testCase.err)
+			}
+			return nil
+		},
+	)
+	if err != nil {
+		log.Fatalln("Error occurred while traversing content:", err)
+	}
+
+	// in case no matched script files were found
+	if len(testCases) == 0 {
+		log.Printf("Warning: no test scripts are found that match '%v'", testsToRun)
+	}
+}
+
+// checkFile takes a file path as the input and returns a TestCase object
+// that is constructed out of the file as a file description
+func checkFile(path string) *TestCase {
+	// check if ends with test.sh and if required to run
+	if !strings.HasSuffix(path, testFileSuffix) {
+		return &TestCase{}
+	} else if !runAllTests && !matched(path, testsAsSlice) {
+		return &TestCase{}
+	}
+	shortPath := path[len(contentFolder):]
+
+	// read the script file
+	script, err := ioutil.ReadFile(path)
+	if err != nil {
+		return &TestCase{path: shortPath, err: err}
+	}
+
+	// parse the script into test and cleanup
+	splitScript := strings.Split(string(script), testCleanupSep)
+	if numParts := len(splitScript); numParts != 2 {
+		err := fmt.Errorf(
+			"script error: expected two-part script separated by '%v', got %v part(s)",
+			testCleanupSep, numParts,
+		)
+		return &TestCase{path: shortPath, err: err}
+	}
+	helperScript := getHelperScript(shortPath)
+	testScript := splitScript[0]
+	cleanupScript := splitScript[1]
+
+	// copy the files sourced by test to cleanup
+	re := regexp.MustCompile("(?m)^source \".*\\.sh\"$")
+	sources := re.FindAllString(testScript, -1)
+	cleanupScript = strings.Join(sources, "\n") + cleanupScript
+
+	// find setup configuration
+	re = regexp.MustCompile(fmt.Sprintf("(?m)^%v (.*)$", setupSpec))
+	setups := re.FindAllStringSubmatch(testScript, -1)
+	if numSetups := len(setups); numSetups != 1 {
+		err := fmt.Errorf(
+			"script error: expected one line that starts with '%v', got %v line(s)",
+			setupSpec, numSetups,
+		)
+		return &TestCase{path: shortPath, err: err}
+	}
+	config := setups[0][1]
+
+	return &TestCase{
+		valid:         true,
+		path:          shortPath,
+		config:        config,
+		testScript:    helperScript + testScript,
+		cleanupScript: helperScript + cleanupScript,
+	}
+}
+
+// NeedSetup checks if any of the test cases require the setup config
+// specified by the input
+func NeedSetup(config string) bool {
+	for idx := range testCases {
+		if testCases[idx].config == config {
+			log.Printf("Setting up istio with %v", config)
+			return true
+		}
+	}
+	log.Printf("No tests need to be run with %v", config)
+	return false
+}
+
+// TestDocs traverses through all test cases and runs those that need the
+// setup config specified by the input. The (*testing.T) variable comes
+// from the TestDocs function in each tests/setup/*/doc_test.go
+func TestDocs(t *testing.T, config string) {
```

Comment:
PascalCase means that a method is public and camelCase means it's private. Let's make sure the functions have the correct casing.

Reply:
These actually have to be public as NeedSetup and TestDocs are called from, say, `tests/setup/profile_default/doc_test.go`, so I'd assume this is okay.